src/main/java/by/custom/utilcalculator/domain/FileUserProgressStorage.java

```diff
@@ -0,0 +1,73 @@
+package by.custom.utilcalculator.domain;
+
+import by.custom.utilcalculator.exception.ReadingUserProgressFromFileException;
+import by.custom.utilcalculator.exception.UserFileNotFoundException;
+import by.custom.utilcalculator.exception.UtilsborException;
+import by.custom.utilcalculator.exception.WritingUserProgressIntoFileException;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class FileUserProgressStorage implements IUserProgressStorage {
+    private final String databaseEnvVariable = System.getenv("CHATBOT_FILE_STORAGE");
+    private final String path = databaseEnvVariable + "%s.dat";
+
+    private static class FileUserProgressStorageHolder {
+        private static final FileUserProgressStorage FILE_USER_PROGRESS_STORAGE = new FileUserProgressStorage();
+    }
+
+    private FileUserProgressStorage() {
+        File databaseDirectory = new File(databaseEnvVariable);
+        if (!databaseDirectory.exists()) {
+            databaseDirectory.mkdirs();
+        }
+    }
+
+    public static FileUserProgressStorage getInstance() {
+        return FileUserProgressStorageHolder.FILE_USER_PROGRESS_STORAGE;
+    }
+
+    @Override
+    public void save(final UserProgress userProgress) throws WritingUserProgressIntoFileException {
+        String chatID = userProgress.getChatID();
+        String filePath = buildFilePath(chatID);
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(filePath))){
+            oos.writeObject(userProgress);
+        } catch (IOException e) {
+            throw new WritingUserProgressIntoFileException(chatID, filePath, e.getCause());
```

Comment:
The cause of `WritingUserProgressIntoFileException` is `IOException` and the cause of IOException is e.getCause(). We must not lose info of IOException.
```suggestion
            throw new WritingUserProgressIntoFileException(chatID, filePath, e);
```

Reply:
It confuses me, but I think, I got it. 
It helps us to get this kind of message: https://c2n.me/4jXbK1z 
If we send just e.getCause() - we have message in stacktrace without Caused by: